app/controllers/api/users_controller.rb

```diff
@@ -2,18 +2,28 @@ class Api::UsersController < ApiController
   # before_action :authenticate
   before_action :authenticated?
 
+  def index 
+    users = User.all
+    render json: users
+  end
+
   def create
     user = User.new(user_params)
     if user.save
-      render json: user
+      render json: user, status: 201
     else
       render json: { errors: user.errors.full_messages }, status: 422
     end
   end
 
-  def index 
-    users = User.all
-    render json: users
+  def destroy
+    begin
+      user = User.find(params[:id])
+      user.destroy
+      render json: {}, status: 204
+    rescue ActiveRecord::RecordNotFound
+      render json: {}, status: 404
+    end
   end
```

Comment:
Nice usage of exceptions (`raise` and `rescue`). Note that in this simple case, you can use a shorter notation:

``` ruby
def destroy
  User.find(params[:id]).destroy
  render json: {}, status: 204
rescue ActiveRecord::RecordNotFound
  render json: {}, status: 404
end
```

Also be aware that if the user can't be destroyed for some reason, the action would silently fail. You can inject and `if`/`else` statement within the `begin`/`rescue` block, or you can also rescue from several types of exceptions:

``` ruby
begin
  # -
rescue OneTypeOfException
  # -
rescue AnotherTypeOfException
  # -
else
  # Handle any other exceptions
end
```

I'm not sure whether in this case, using Ruby 2.*, you can use the shorter notation, though (an explicit `begin`/`rescue`/`end` block may be required).